pkg/storer/internal/upload/uploadstore.go

```diff
@@ -474,168 +465,117 @@ func (u *uploadPutter) Close(s internal.Storage, writer storage.Writer, addr swa
-		if err != nil {
-			return fmt.Errorf("failed reading dirty tag while cleaning up: %w", err)
-		}
-
-		return s.IndexStore().Iterate(
-			storage.Query{
-				Factory:       func() storage.Item { return &pushItem{} },
-				PrefixAtStart: true,
-				Prefix:        fmt.Sprintf("%d", di.Started),
-			},
-			func(res storage.Result) (bool, error) {
-				pi := res.Entry.(*pushItem)
-				if pi.TagID == u.tagID {
-					itemsToDelete = append(itemsToDelete, pi)
-				}
-				return false, nil
-			},
-		)
-	})
+	di := &dirtyTagItem{TagID: u.tagID}
+	err := st.IndexStore().Get(di)
 	if err != nil {
-		return fmt.Errorf("failed iterating over push items: %w", err)
+		return fmt.Errorf("failed reading dirty tag while cleaning up: %w", err)
 	}
 
-	batchCnt := 1000
-	for i := 0; i < len(itemsToDelete); i += batchCnt {
-		err = tx.Execute(context.Background(), func(st internal.Storage) error {
-
-			b, err := st.IndexStore().Batch(context.Background())
-			if err != nil {
-				return err
-			}
-
-			end := i + batchCnt
-			if end > len(itemsToDelete) {
-				end = len(itemsToDelete)
-			}
-			for _, pi := range itemsToDelete[i:end] {
-				_ = remove(st, b, pi.Address, pi.BatchID)
-				_ = b.Delete(pi)
+	err = st.IndexStore().Iterate(
+		storage.Query{
+			Factory:       func() storage.Item { return &pushItem{} },
+			PrefixAtStart: true,
+			Prefix:        fmt.Sprintf("%d", di.Started),
+		},
+		func(res storage.Result) (bool, error) {
+			pi := res.Entry.(*pushItem)
+			if pi.TagID == u.tagID {
+				itemsToDelete = append(itemsToDelete, pi)
 			}
-			return b.Commit()
-		})
-		if err != nil {
-			return fmt.Errorf("failed deleting push items: %w", err)
-		}
-	}
-
-	return tx.Execute(context.Background(), func(tx internal.Storage) error {
-		return tx.IndexStore().Delete(&dirtyTagItem{TagID: u.tagID})
-	})
-}
-
-// Remove removes all the state associated with the given address and batchID.
-func remove(st internal.Storage, writer storage.Writer, address swarm.Address, batchID []byte) error {
-	ui := &uploadItem{
-		Address: address,
-		BatchID: batchID,
-	}
-
-	err := st.IndexStore().Get(ui)
-	if err != nil {
-		return fmt.Errorf("failed to read uploadItem %s: %w", ui, err)
-	}
-
-	err = writer.Delete(ui)
+			return false, nil
+		},
+	)
 	if err != nil {
-		return fmt.Errorf("failed deleting upload item: %w", err)
+		return fmt.Errorf("failed iterating over push items: %w", err)
 	}
 
-	if ui.Synced == 0 {
-		err = st.ChunkStore().Delete(context.Background(), address)
-		if err != nil {
-			return fmt.Errorf("failed deleting chunk: %w", err)
-		}
-
-		stamp, err := chunkstamp.LoadWithBatchID(st.IndexStore(), chunkStampNamespace, address, batchID)
-		if err != nil {
-			return fmt.Errorf("failed getting stamp: %w", err)
-		}
-
-		err = chunkstamp.DeleteWithStamp(writer, chunkStampNamespace, address, stamp)
-		if err != nil {
-			return fmt.Errorf("failed deleting chunk stamp %x: %w", batchID, err)
-		}
-	}
-	return nil
+	var eg errgroup.Group
+	eg.SetLimit(8)
```

Comment:
8 seems like a value that is optimal for some systems, others may be better with different constant. Is there a sense to put the number of CPUs?

Reply:
the idea here is to prevent the loop from creating too many goroutines and overload the cpu/disk

Reply:
Exactly because of this. Value 8 may be good for some system, while other values can be good for other systems. Like where is only one CPU, the value of 1 would probably be better than 8. That is why I've suggested to set the number of CPUs instead the fixed value of 8.

Reply:
aha I see what you mean, then i will use `runtime.NumCPU()`